cas-server-support-pac4j/src/main/java/org/apereo/cas/support/pac4j/web/flow/ClientAction.java

```diff
@@ -176,7 +157,7 @@ protected void prepareForLoginPage(final RequestContext context) {
             final IndirectClient indirectClient = (IndirectClient) client;
             // clean Client suffix for default names
             final String name = client.getName().replace("Client", "");
-            final String redirectionUrl = indirectClient.getRedirectionUrl(webContext);
+            final String redirectionUrl = indirectClient.getRedirectAction(webContext).getLocation();
```

Comment:
Here is my real concern. This is the comeback of the hardly-understandable concept of direct / indirect redirection.

If you do that, it will work. But you may encounter performance issues. If you choose Twitter for example (which is based on OAuth 10), the computation of the redirection url requires to call the Twitter server. So if it is down or slow, you will have troubles to display the login page even if you don't click on the "Twitter login" button.

I think an intermediate url (/delegateauthentication?client_name=TwitterClient) would be necessary to avoid this downside.